python/cudf/cudf/core/column/struct.py

```diff
@@ -153,6 +154,36 @@ def _with_type_metadata(self: StructColumn, dtype: Dtype) -> StructColumn:
 
         return self
 
+    def serialize(self) -> Tuple[dict, list]:
+        header, frames = super().serialize()
+        header["dtype"] = self.dtype.serialize()
+        header["size"] = self.size
+
+        header["sub-frame-offset"] = len(frames)
```

Comment:
There doesn't appear to be much detail, since `serialize/deserialize` are effectively just ad-hoc polymorphic the interpretation of headers and frames is just left up to the pairs for each type.

Reply:
Okay, that sounds in line with my understanding of the serialization. I do think that the sub-frame offset is a non-obvious part of the struct serialization method. Just a one-line comment would help, if this is accurate:
```suggestion
        # The sub-frame-offset denotes the frame index where parent column
        # data ends and child column data begins.
        header["sub-frame-offset"] = len(frames)
```